app/models/page_template.rb

```diff
@@ -1,6 +1,19 @@
 # Model for page templates
 class PageTemplate < ApplicationRecord
-  # Associations
+  validates :name,     presence: true
+  validates :filename, presence: true
+
   has_many :pages, foreign_key: 'template_id', inverse_of: 'template',
                    dependent: :delete_all
+
+  # Instance methods
+
+  # Check whether the template file is present on disk
+  def file_exists?
+    File.exist?(
+      Rails.root.join(
+        'app', 'views', 'pages', 'templates', "#{filename}.html.erb"
```

Comment:
I suspect there may be a vulnerability here that allows file traversal if the submitted filename `contains` "../../".

Consider applying a strict safelist, where only the filenames in `app/views/pages/templates/*.html.erb` are allowed, rejecting anything that doesn't match one of those filenames exactly.

`Dir.glob` is one option for implementing this: https://ruby-doc.org/core-2.6.3/Dir.html#method-c-glob